libs/blocks/bulk/utils.js

```diff
@@ -0,0 +1,246 @@
+import { getImsToken } from '../../../tools/send-to-caas/send-utils.js';
+import { fetchWithTimeout } from '../caas/utils.js';
+import { loadScript } from '../../utils/utils.js';
+import {
+  getLocalStorage,
+  setLocalStorage,
+} from '../review/utils/localStorageUtils.js';
+
+export const ADMIN_BASE_URL = 'https://admin.hlx.page';
+const THROTTLING_DELAY_MS = 100;
+export const BULK_CONFIG_FILE_PATH = '/tools/bulk-config.json';
+export const BULK_REPORT_FILE_PATH = '/tools/bulk-report';
+const BULK_AUTHORIZED_USERS = 'bulkAuthorizedUsers';
+export const BULK_SUPPORTED_SITES = 'bulkSupportedSites';
+const BULK_STORED_URL_IDX = 'bulkStoredUrlIdx';
+const BULK_STORED_COMPLETED = 'bulkStoredCompleted';
+const BULK_STORED_URLS = 'bulkStoredUrls';
+const BULK_STORED_OPERATION = 'bulkStoredOperation';
+const UNSUPPORTED_SITE_STATUS = 'unsupported domain';
+
+// eslint-disable-next-line no-promise-executor-return
+const delay = (milliseconds) => new Promise((resolve) => setTimeout(resolve, milliseconds));
+
+export const getUser = async () => {
+  const profile = await window.adobeIMS?.getProfile();
+  return profile
+    ? { name: profile.name, email: profile.email }
+    : { name: 'anonymous', email: 'anonymous' };
+};
+
+export const signIn = async () => {
+  // force user to sign in
+  const accessToken = await getImsToken(loadScript);
+  if (!accessToken) {
+    window.adobeIMS.signIn();
+    return false;
+  }
+  return true;
+};
+
+export const signOut = (e) => {
+  e.preventDefault();
+  window.adobeIMS?.signOut();
+};
+
+export const getAuthorizedUsers = async () => {
```

Comment:
Security on the frontend? There is none. 
But this is good enough given the use-case?